miraclelinux/meta-emlinux
# linux: Add security hardened configuration

# Purpose of pull request

Add security hardened configuration file.
When CONFIG_EXPERT set to "y", some values were changed. Therefore it needs to revert this side effect.

# Test
## How to test

e.g. 

Set following value to conf/local.conf
```
EMLINUX_SECURITY_HARDEND="1"
```

Then build image and run it.

```
$ bitbake core-image-minimal 
$ runqemu qemuarm64
```

Then, check some config values in /proc/config.gz.

## Test result

qemu gui still work.

![Screenshot from 2021-09-29 14-42-17](https://user-images.githubusercontent.com/165052/135217239-bf74cd5e-0ea7-44b5-8a28-9c07854f0206.png)

configurations are correct as expected.

![Screenshot from 2021-09-29 14-44-53](https://user-images.githubusercontent.com/165052/135217258-49ae53ef-d22e-4e41-b74c-d15abc209efb.png)
